### ISSUE TITLE
Support additional resource properties with `add_resource(...)`

### ISSUE THREAD

**Issue by peterdesmet** [frictionlessdata/frictionless-r#140]
`add_resource()` will [add](https://github.com/frictionlessdata/frictionless-r/blob/b0a9fe34b26fe120309899503bba67cffcb710ff/R/add_resource.R#L116) the following properties to resources:

- name
- data or path
- profile
- schema
- format (when path is used)
- mediatype (when path is used)
- encoding (when path is used)
- dialect (when path is used + dialect is not default)

But resources can have [additional properties](https://specs.frictionlessdata.io/data-resource/#optional-properties):

- `title`
- `description`
- `bytes`
- `hash`
- `licenses`

Rather than adding all those as extra parameters in `add_resource()`, they could be supported with ellipsis `...`. See the suggestion by @hansvancalster at https://github.com/frictionlessdata/frictionless-r/issues/48#issuecomment-1594399115

**Comment by peterdesmet** [comment]
Note that there are only 5 unsupported additional properties, so we could be defensive and hardcode those. On the other hand, the Frictionless specs do allow to expand with your own properties, so I still think that `...` is a good approach. But it will be up to the user then, that e.g. adding a property `hash` or `bytes` meets the requirements of the definition.

I would return a message if custom properties are defined:

> The following custom properties were provided:
> ✖ `path`: ignored, is set by the function.
> ℹ `sources`: added, should meet requirements defined at https://specs.frictionlessdata.io/data-resource/#optional-properties.
> ✅ `blabla`: added.

That message could be silenced by `suppressMessages()`

**Comment by peterdesmet** [comment]
Note to self: implementation in #141 returns:


``` r
library(frictionless)
p <- create_package()
p1 <- add_resource(p, "mtcars", mtcars)
p2 <- add_resource(p, "mtcars", mtcars, foo = "bar")
#> The following custom properties were provided:
#>   foo: added.
p3 <- add_resource(p, "mtcars", mtcars, source = "test")
#> The following custom properties were provided:
#>   source: added.
p4 <- add_resource(p, "mtcars", mtcars, path = "test")
#> Error: path are passed to ... but are set internally. Please remove them.
```

<sup>Created on 2023-06-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by peterdesmet** [comment]
On second inspection, I think it is better:

- Not to return a message. This is unnecessarily verbose for a functionality that will mostly be used by other functions.
- Not to differentiate between custom (`foo`) and defined properties (`title`). The documentation indicates that properties are not verified.
- Return an error for properties that are reserved. Developers of other functions that make use of `...` should be warned that they have no control over these properties.

**Comment by peterdesmet** [comment]
Implemented with #141

**Comment by hansvancalster** [comment]
Great! Thanks for the swift follow-up and for merging #141.
